crev-lib/src/util/git.rs

```diff
@@ -34,14 +35,40 @@ pub fn parse_git_url_https(http_url: &str) -> Option<GitUrlComponents> {
 }
 
 pub fn fetch_and_checkout_git_repo(repo: &git2::Repository) -> Result<()> {
-    repo.find_remote("origin")?.fetch(&["master"], None, None)?;
+    let mut fetch_options = default_fetch_options();
+    repo.find_remote("origin")?
+        .fetch(&["master"], Some(&mut fetch_options), None)?;
     repo.set_head("FETCH_HEAD")?;
     let mut opts = git2::build::CheckoutBuilder::new();
     opts.force();
     repo.checkout_head(Some(&mut opts))?;
     Ok(())
 }
 
+/// Make a git clone with the default fetch options
+pub fn clone<P: AsRef<Path>>(
+    url: &str,
+    path: P,
+) -> std::result::Result<git2::Repository, git2::Error> {
+    let fetch_options = default_fetch_options();
+    git2::build::RepoBuilder::new()
+        .fetch_options(fetch_options)
+        .clone(url, path.as_ref())
+}
+
+/// Get the default fetch options to use when fetching or cloneing
+///
+/// Currently this just ensures that git's automatic proxy settings are used.
+pub fn default_fetch_options<'a>() -> git2::FetchOptions<'a> {
```

Comment:
Wait, what? Where's this `<'a>` coming from? :D

Reply:
That is a good question, but for some strange reason the [`git2::FetchOptions`](https://docs.rs/git2/0.13.1/git2/struct.FetchOptions.html) acutally has a lifetime named `'dp` in it and I don't know where it comes from. If I don't put the `'a` in the function, Rust complains that there is nothing to borrow the `FetchOptions` from.